src/app/common/stores/applications/SsvWeb/Validator.store.ts

```diff
@@ -152,38 +150,31 @@ class ValidatorStore extends BaseStore {
   }
 
   async removeValidator(publicKey: string, operatorIds: number[]): Promise<boolean> {
-    const contract = getContractByName(EContractName.SETTER);
-    store.dispatch(setIsLoading(true));
     const isContractWallet = store.getState().walletState.isContractWallet;
     const accountAddress = store.getState().walletState.accountAddress;
     const ssvStore: SsvStore = this.getStore('SSV');
-    const sortedOperatorIds = getSortedOperatorsIds(operatorIds);
-    const clusterData = await getClusterData(getClusterHash(operatorIds, accountAddress), ssvStore.liquidationCollateralPeriod, ssvStore.minimumLiquidationCollateral);
     return new Promise(async (resolve) => {
       try {
-        const tx = await contract.removeValidator(publicKey, sortedOperatorIds, clusterData);
-        if (tx.hash) {
-          store.dispatch(setTxHash(tx.hash));
-          store.dispatch(setIsShowTxPendingPopup(true));
-        }
-        if (isContractWallet) {
-          resolve(true);
-        }
-        const receipt = await tx.wait();
-        if (receipt.blockHash) {
-          await executeAfterEvent(async () => !await getValidator(publicKey), async () => this.myAccountStore.refreshOperatorsAndClusters(), delay);
-          resolve(true);
-        } else {
+        const sortedOperatorIds = getSortedOperatorsIds(operatorIds);
+        const clusterData = await getClusterData(getClusterHash(operatorIds, accountAddress), ssvStore.liquidationCollateralPeriod, ssvStore.minimumLiquidationCollateral);
+        const payload = { publicKey, sortedOperatorIds, clusterData };
+        const contract = getContractByName(EContractName.SETTER);
+        if (!payload) {
           resolve(false);
+          return;
         }
+        await transactionExecutor({
+          methodType: TransactionMethod.RemoveValidator,
+          payload,
+          contract,
+          isContractWallet: isContractWallet,
+          callbackAfterExecution: async () => await this.myAccountStore.refreshOperatorsAndClusters(),
```

Comment:
pass pointer to function 
`callbackAfterExecution: this.myAccountStore.refreshOperatorsAndClusters`